packages/grafana-data/src/utils/numbers.ts

```diff
@@ -0,0 +1,29 @@
+/**
+ * Round half away from zero ('commercial' rounding)
+ * Uses correction to offset floating-point inaccuracies.
+ * Works symmetrically for positive and negative numbers.
+ *
+ * ref: https://stackoverflow.com/a/48764436
+ */
+export function roundDecimals(val: number, dec = 0) {
+  if (Number.isInteger(val)) {
+    return val;
+  }
+
+  let p = 10 ** dec;
+  let n = val * p * (1 + Number.EPSILON);
+  return Math.round(n) / p;
+}
+
+/**
+ * Tries to guess number of decimals needed to format a number
+ *
+ * used for determining minimum decimals required to uniformly
+ * format a numric sequence, e.g. 10, 10.125, 10.25, 10.5
+ *
+ * good for precisce increments:  0.125            -> 3
+ * bad  for arbitrary floats:     371.499999999999 -> 12
+ */
+export function guessDecimals(num: number) {
+  return (('' + num).split('.')[1] || '').length;
+}
```

Comment:
**Suggestion:** I think we should add some tests for these utility functions, especially as they are used in a lot of places.

Although I like the comments here (they are making the "why" easier to understand), tests could also help making it easier to understand what these are doing (apart from being a safety net).